test/integration/CodeExamples.test.ts

```diff
@@ -676,6 +676,49 @@ describe('Marketplace flow tests', async () => {
     }
   }) ///
   /// ```
+
+  it('8.1 Add key-value pair to data NFT', async () => {
+    /// Data NFTs can store arbitrary key-value pairs on-chain. This opens up their usage for a broad variety of applications, such as comments & ratings, attestations, and privately sharing data (when the value is encrypted).
+
+    /// Let's see how!
+
+    /// Here are the steps:
+
+    /// 1. Setup (same as above)
+    /// 2. Publish data NFT (same as above)
+    /// 3. Add key-value pair to data NFT (use the `setData` method)
+    /// 4. Retrieve value from data NFT (use the `getData` method)
+
+    /// Let's start by using the `setData` method to update the nft key value store with some data
+    /// ```Typescript
+    const nft = new Nft(web3)
+    const data = 'SomeData'
+    try {
+      await nft.setData(freNftAddress, publisherAccount, '1', data)
+    } catch (e) {
+      console.error(e)
+      assert.fail('Download failed')
```

Comment:
we need a proper assertion fail message here pointing to the real possible issue on setting nft data